src/pages/SignUp.js

```diff
@@ -0,0 +1,89 @@
+import { useEffect } from "react"
+import { useSelector, useDispatch } from "react-redux"
+import { useNavigate } from "react-router-dom"
+import { toast } from "react-toastify"
+import { signUp, reset } from "../features/auth/authSlice"
+import DefaultContainer from "../layout/DefaultContainer"
+import { StyledInput, PasswordInput } from "../elements/Input"
+import { Button } from "../elements/Button"
+import Spinner from "../layout/Spinner"
+import useForm from "../hooks/useForm"
+
+function SignUp() {
+  const { formData, onChange } = useForm({
+    username: "",
+    email: "",
+    password: "",
+    password2: "",
+  })
+  const { password, password2 } = formData
+  const navigate = useNavigate()
+  const dispatch = useDispatch()
+
+  const { user, isLoading, isSuccess, isError, message } = useSelector(
+    (state) => state.auth
+  )
+
+  useEffect(() => {
+    if (isError) {
+      toast.error(message)
+    }
+    if (isSuccess || user) {
+      navigate("/")
+    }
+    dispatch(reset())
+  }, [user, isError, isSuccess, message, navigate, dispatch])
+
+  const onSubmit = (e) => {
+    e.preventDefault()
+
+    if (password !== password2) {
```

Comment:
Better here use `formData.password` directly, so we don't create too much variables